src/redux/saga/adItemSaga.ts

```diff
@@ -0,0 +1,17 @@
+import { put } from 'redux-saga/effects';
+import { addItemAction } from '../actions/todoActions/addItemAction';
+import { createReferenceHelper } from '../../helpers/createReferenceHelper';
+import { AddItemSagaAction } from '../actions/todoSagaActions/addItemSagaAction';
+
+export function* addItemSaga(action: AddItemSagaAction) {
+  const { newItem, userToken } = action.payload;
+  try {
+    yield createReferenceHelper
+      .ref(`Users/${userToken}/Todo/`)
+      .child(`${newItem.id}`)
+      .set(newItem);
+    yield put(addItemAction({ newItem }));
+  } catch (error) {
+    console.error(error);
```

Comment:
let's make native alerts in catch, like title: "Something went wrong", description from firebase error or you own error("We couldn't add item, please try again").

Make simple function(helper) that will get string and return alert with title and description

Reply:
in other catches further do the same

Reply:
will be done